pulled out FilterVariants Expr and List functions

jbloom22 opened the pull request:
to facilitate calling these functions from elsewhere in Hail without changing vds in state, which I need for linear mixed models

tpoterba commented on src/main/scala/org/broadinstitute/hail/driver/FilterVariantsExpr.scala:
```diff
@@ -40,10 +38,13 @@ object FilterVariantsExpr extends Command {
+    state.copy(vds = filterVariantsExpr(vds, cond, keep))
+  }
+
+  def filterVariantsExpr(vds: VariantDataset, cond: String, keep: Boolean): VariantDataset = {
```
I think this might fit better as a method on VSM:
```scala
state.copy(vds = vds.filterVariantsExpr(cond, keep))
```

Should be easy to move it there.

tpoterba commented on src/main/scala/org/broadinstitute/hail/driver/FilterVariantsList.scala:
```diff
@@ -35,10 +35,13 @@ object FilterVariantsList extends Command {
+    state.copy(vds = filterVariantsList(vds, options.input, options.keep))
+  }
+
+  def filterVariantsList(vds: VariantDataset, input: String, keep: Boolean): VariantDataset = {
```
Same with this one.

jbloom22 commented:
Good idea Tim. I moved List to VSM and Expr to RichVDS due to this line which requires vds:
`val aggregatorOption = Aggregators.buildVariantAggregations(vds, aggregationEC)`

I also moved the "build the variant rdd" bit to Variant as that seems useful more generally. Back to you.
